API.md

```diff
@@ -176,212 +90,777 @@ This page documents the Party service API endpoints. All endpoints return an `HT
+    * Sends password reset link to the provided email address.
+    * Raises an error if email not found in DB.
+    * Requires `email_address` param.
+
+### Example JSON Payload
+```json
+[
+  {
+  "email_address" : "a@z.com"
+  }
+]
+```
 
 ### Example JSON Response
 ```json
 {
-    "collectionExerciseId": "aCollectionExerciseId",
-    "sampleSummaryId": "aSampleSummaryId"
+  "response" : "ok"
 }
 ```
+---
+### Create New Party
+* `POST /party-api/v1/respondents`
+    * Posts a respondent to the database and generates their `sampleUnitRef`, `partyID` and `enrolments`.
+    * If passed an `id` parameter it will use this instead of generating a new UUID.
+    * Sets `businessRespondentStatus` to 'CREATED'.
+
+### Example JSON Payload
+```json
+[
+  {
+    "emailAddress" : "example@example.com",
+    "firstName" : "Bob",
+    "lastName" : "Dabilder",
+    "password" : "password",
+    "telephone" : "01234567890",
+    "enrolmentCode" : "<enrolmentCode>",
+    "sampleUnitType" : "BI"
+  }
+]
 
-* `POST /respondents/add-survey`
+```
+---
+### Verify Email Address with Token
+* `PUT /emailverification/<token>`
+    * Verifies the users email address against a provided token.
+    * Example token: `'ImFAei5jb20i.W-7Ovg.hFZ7nhkzq8e7i76EXSwgvJQXAjs'`
+    
+### Example JSON Response
+```json
+{
+  "id": "<id>",
+  "sampleUnitType": "BI",
+  "pendingEmailAddress": "",
+  "emailAddress": "a@z.com",
+  "firstName": "Hollie",
+  "lastName": "Day",
+  "telephone": "01234567890",
+  "status": "ACTIVE",
+  "associations": [
+        {
+        "enrolments": [
+            {
+                "enrolmentStatus": "ENABLED",
+                "surveyId": "<surveyId>"
+            }
+        ],
+        "partyId": "<partyId>",
+        "businessRespondentStatus": "ACTIVE"
+        }
+    ]
+        }
+```
 
-### Example JSON data for post
+---
+### Re-send Verification Email
+* `POST /resend-verification-email/<party_uuid>`
+    * Sends another email containing a verification token.
+    * Raises an error if email is not found in DB.
+    
+### Example JSON Response
 ```json
 {
-    "party_id": "438df969-7c9c-4cd4-a89b-ac88cf0bfdf3",
-    "enrolment_code": "fb747cq725lj"
+  "message": "A new verification email has been sent"
 }
 ```
-&mdash; This endpoint will enrol a respondent in a survey and associate with business if not already associated.
 
+---
+### Re-send Verification Email (Token Expired)
+* `POST /resend-verification-email-expired-token/<token>`
+    * Allows an internal user to send another verification email to respondent.
 
-* `PUT /respondents/edit-account-status/<respondent_party_id>`
+### Example JSON Response
+```json
+{
+  "message": "A new verification email has been sent"
+}
+```
+---
+### Re-send Password Email (Token Expired)
+* `POST /resend-password-email-expired-token/<token>`
+    * Sends the respondent another password verification URL containing a token via email.
+### Example JSON for Post
+```json
+{
+  "email_address" : "a@b.com"
+}
```

Comment:
This endpoint doesn't need any JSON to be passed to it- it works out the email address by decoding the token you give it.